### PR TITLE
Generalize reloadableDomains

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

This cuts out the manual having to add domains to the `reloadableDomains` array and having to add translations for every domain. The default translation can still be overwritten.

Also allows custom integrations to show up in the reload list.

/CC @bdraco 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by balloob — APPROVED** [review]

File: src/translations/en.json [hunk 1]
```diff
@@ -827,32 +827,8 @@
...
-              "filter": "Reload filter entities",
-              "statistics": "Reload statistics entities",
-              "generic": "Reload generic IP camera entities",
-              "generic_thermostat": "Reload generic thermostat entities",
-              "homekit": "Reload HomeKit",
-              "min_max": "Reload min/max entities",
-              "history_stats": "Reload history stats entities",
-              "trend": "Reload trend entities",
-              "ping": "Reload ping binary sensor entities",
-              "filesize": "Reload file size entities"
```
Comment (ludeeus) [inline src/translations/en.json:855]: Should we keep the translations that do not result in an identical string?

Comment (bramkragten) [reply src/translations/en.json:855]: Yeah we can, should we leave all in? They all are slightly different I think, mainly because of plurals

Comment (ludeeus) [reply src/translations/en.json]: yeah, why not.
We can always change it / remove some in the future.

**Review by ludeeus — APPROVED** [review]

File: src/translations/en.json (hunk 1 above)
Comment (Misiu) [reply src/translations/en.json]: @bramkragten if possible please add rpi_gpio to the translations (https://github.com/home-assistant/frontend/pull/6764/files#diff-571ce32a17d6aa22fdd0942224a8ff44R856)

`"rpi_gpio": "Reload Raspberry Pi GPIO entities"`

Comment (bramkragten) [reply src/translations/en.json]: What does it say now?

Comment (bramkragten) [reply src/translations/en.json]: If it needs to change, please create a PR

Comment (Misiu) [reply src/translations/en.json]: currently, there is no rpi_gpio entry.
I'll check how this looks and if needed create a new PR.

**Pull request merged**